Source_Files/Misc/preferences.cpp

```diff
@@ -1265,11 +1272,18 @@ static void graphics_dialog(void *arg)
 	table->dual_add_row(new w_static_text("*may interfere with third-party scenario effects"), d);
 
 	table->add_row(new w_spacer(), true);
-	
-	w_toggle *bob_w = new w_toggle(graphics_preferences->screen_mode.camera_bob);
-	table->dual_add(bob_w->label("Camera Bobbing"), d);
-	table->dual_add(bob_w, d);
 
+	w_select *bob_disable_w = new w_select(0, bob_disable_labels);
+	for (auto i = 0; bob_disable_labels[i] != NULL; ++i)
+	{
+		if (bob_disable_values[i] == graphics_preferences->screen_mode.bob_disable)
+		{
+			bob_disable_w->set_selection(i);
+		}
+	}
```

Comment:
Since the array values are the same as the array indexes, you can directly call `set_selection` with the bob_disable value and don't need for a for/if.